bundles/org.openhab.core.model.yaml/src/main/java/org/openhab/core/model/yaml/YamlModelListener.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.model.yaml;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link YamlModelListener} interface is responsible for managing a particular model type
+ * with data processed from YAML configuration files.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+public interface YamlModelListener<T extends YamlElement> {
+
+    /**
+     * Method called by the model repository when elements from a model are added.
+     *
+     * @param modelName the name of the model
+     * @param elements the list of added elements
+     */
+    void addedModel(String modelName, List<? extends YamlElement> elements);
+
+    /**
+     * Method called by the model repository when elements from a model are updated.
+     *
+     * @param modelName the name of the model
+     * @param elements the list of updated elements
+     */
+    void updatedModel(String modelName, List<? extends YamlElement> elements);
```

Comment:
Just a suggestion: if ordering is not of importance (which is probably true), you could use `Collection<? extends YamlElement>` instead of `List`. The same applies to `getElements` in `YamlFile`. 

In fact I think that in most cases a `Set` might be a more useful implementation than a `List`, I can't think of a reason for non-unique elements. But if the interface is clean, that can be changed without breaking anything.

Reply:
Yes, I think we can use Collection and then I will sort in the implementation.

Reply:
Done